apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerRequestId.cs

```diff
@@ -0,0 +1,168 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading;
+using Google.Cloud.Spanner.V1.Internal.Logging;
+
+namespace Google.Cloud.Spanner.V1;
+
+/// <summary>
+/// Represents a structured request ID for Spanner RPCs, formatted as:
+/// {version}.{process}.{client}.{channel}.{request}.{attempt}
+/// </summary>
+internal sealed class SpannerRequestId
+{
+    /// <summary>
+    /// The header key used for Spanner request IDs.
+    /// </summary>
+    internal const string HeaderKey = "x-goog-spanner-request-id";
+
+    private static Lazy<string> s_process = new(GenerateProcessId);
+    /// <summary>
+    /// A unique ID for the user application process. This can be overridden, but only before it has been accessed.
+    /// </summary>
+    internal static string ProcessId
+    {
+        get => s_process.Value;
+        set
+        {
+            if (s_process.IsValueCreated)
+            {
+                Logger.DefaultLogger.Warn("ProcessId cannot be changed after it has been accessed. The new value will be ignored.");
+            }
+            else
+            {
+                s_process = new Lazy<string>(() => value);
+            }
+        }
+    }
```

Comment:
I think there's a safer way to do this and that is to have the `GenerateProcessId` method interlocked check the value of a settable process ID. If it's already set by the time `GenerateProcessId` checks, then that's used, otherwise a value is generated.